server/src/main/java/com/epam/aidial/core/server/service/PublicationService.java

```diff
@@ -552,6 +601,41 @@ private void copySourceToReviewResources(List<Publication.Resource> resources) {
         }
     }
 
+    private void replaceCustomAppFiles(Application application, Map<String, String> replacementLinks) {
+        if (application.getCustomAppSchemaId() == null) {
+            return;
+        }
+        JsonNode customProperties = ProxyUtil.MAPPER.convertValue(application.getCustomProperties(), JsonNode.class);
+        replaceLinksInJsonNode(customProperties, replacementLinks, null, null);
+        Map<String, Object> customPropertiesMap = ProxyUtil.MAPPER.convertValue(customProperties, new TypeReference<>() {
+        });
+
+        application.setCustomProperties(customPropertiesMap);
+    }
+
+    private void replaceLinksInJsonNode(JsonNode node, Map<String, String> replacementLinks, JsonNode parent, String fieldName) {
```

Comment:
Move all logic to CustomAppUtil and test it

Reply:
moved